Css reset

wackyleo459 opened the pull request:
Issue discussed here: https://github.com/dfinity/internet-identity/issues/785

# Changes
- Reset styling for different browser styles, by using "the-new-css-reset" stylesheet that gets copied into main.css.

- `main.css` has a few added styling to restore some large discrepancies from previous style.

meodai commented:
Nice job! I like that you choose to add it as a dependency.

meodai commented on src/frontend/src/styles/main.css:
```diff
@@ -156,9 +157,9 @@ button {
-
   line-height: 1.25rem;
   letter-spacing: 0.005em;
+  font-family: Arial;
```
I know that it was `Arial` before. But I think it look nicer if its the same font then the rest of the site no? I think it was an omission before.

wackyleo459 replied:
Yup, sure it seems to look great with same font. I might have to adjust certain button font-size, but that should be simple enough!

<image src="https://user-images.githubusercontent.com/38238542/182981797-ea9c3612-4302-40fc-91b8-4a971cbf5f67.jpeg" width=400>

meodai commented:
I have chosen to add the reset within the `main.css` in my branch:
https://github.com/dfinity/internet-identity/blob/feature/css-cleanup/src/frontend/src/styles/main.css

There are some reasons why I think this make sense: 
- I don't see an other way to follow the ITCSS order: https://csswizardry.com/2018/11/itcss-and-skillshare/ (to reconsider if we ever introduce SCSS or/and PostCSS)
- security: its one less npm dependency that could potentially cause harm to the end-user.

Do you think that would also be a good idea in this branch?

wackyleo459 commented:
Yes I have followed your suggestion to include the reset inside the main.css. However, my format is a bit different in that I added the reset styling before everything else. I hope when you eventually merge your branch it will be somewhat straight forward!
@meodai

wackyleo459 commented:
@frederikrothenberger I noticed selenium tests kept failing, and it seemed to be coming from an extra space for a test. Once I fixed that, the tests all passed!

frederikrothenberger commented:
> @frederikrothenberger I noticed selenium tests kept failing, and it seemed to be coming from an extra space for a test. Once I fixed that, the tests all passed!

It seems fine to remove it. Do you know, why this space was there previously (and now isn't)?

frederikrothenberger commented:
This has broken the layout on the remote device flow instructions, e.g. here: [screenshots/09-new-device-instructions-mobile.png](https://github.com/dfinity/internet-identity/pull/796/files#diff-aa730327da063680d3a9c8a61abdd87d9e5cb9d068e88b3326e0b16b69041f3a)

wackyleo459 commented:
> > @frederikrothenberger I noticed selenium tests kept failing, and it seemed to be coming from an extra space for a test. Once I fixed that, the tests all passed!
> 
> It seems fine to remove it. Do you know, why this space was there previously (and now isn't)?

The space seemed to always be there, since when you first created the test. I'm not sure why tests were passing before and now were failing. 
I've updated the styling again, and styling should be consistent regarding numbers and bullets!

nmattia reviewed (commented):
Thanks a lot! 

I like having the reset in the `main.css` as well, for the reasons mentioned by @meodai .

I noticed a few things that broke:

* The "about" compatibility box layout is now vertical whereas it was horizontal before
* Links in both "about" and "faq" aren't styled anymore
* Headers on the compatibility page have lost their weight

Would be great if we could at least fix the headers and links!

wackyleo459 commented:
> Thanks a lot!
> 
> I like having the reset in the `main.css` as well, for the reasons mentioned by @meodai .
> 
> I noticed a few things that broke:
> 
> * The "about" compatibility box layout is now vertical whereas it was horizontal before
> * Links in both "about" and "faq" aren't styled anymore
> * Headers on the compatibility page have lost their weight
> 
> Would be great if we could at least fix the headers and links!

@nmattia These issues have been addressed and fixed in the latest changes. If all tests seem to pass, I hope this PR will go through! =)

nmattia reviewed (approved):
This is great, thanks for addressing those few issues!
